src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Common.targets

```diff
@@ -44,7 +44,23 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <PropertyGroup>
     <IncludeBuildOutput Condition=" '$(PackAsTool)' == 'true' ">false</IncludeBuildOutput>
-    <PackageType Condition=" '$(PackAsTool)' == 'true' ">DotnetTool</PackageType>
+    <!--
+    PackageType is semicolon-delimited, case-insensitive, with an optional version.
+
+    DotnetTool should be added to the set of package types if the PackAsTool property is set to true.
+
+    Examples:
+       PackageType = ''                     -> 'DotnetTool'
+       PackageType = 'MyCustomType'         -> 'DotnetTool;MyCustomType'
+       PackageType = 'dotnettool'           -> 'dotnettool'
+       PackageType = 'DotnetTool, 1.0.0.0'  -> 'DotnetTool, 1.0.0.0'
+       PackageType = 'DotnetTool , 1.0.0.0' -> 'DotnetTool , 1.0.0.0'
+    
+    _PaddedPackageType is used to ensure that the PackageType is semicolon delimited and can be easily checked for an existing DotnetTool package type.
+    -->
+    <_PaddedPackageType Condition=" '$(PackAsTool)' == 'true' ">;$(PackageType.Replace(' ', '').Trim().ToLowerInvariant());</_PaddedPackageType>
+    <PackageType Condition=" '$(PackAsTool)' == 'true' and '$(_PaddedPackageType)' != ';;' and !$(_PaddedPackageType.Contains(';dotnettool;')) and !$(_PaddedPackageType.Contains(';dotnettool,')) ">DotnetTool;$(PackageType)</PackageType>
```

Comment:
You can use ordinal ignore case in the comparers as well. 

> _PaddedPackageType.Contains(';dotnettool,', StringComparison.OrdinalIgnoreCase)

That being said, not sure that becomes much simpler, since you're now account for many scenarios, like say package types start with the same prefix. 

idk if you need to teheoretically check against:

`,dotnettool;` as well.

I think package types are `;` delimited only so maybe you don't need the other `,` check.

Reply:
Since the approach uses a new property anyways and there are two Contains checks, I think it is simpler to normalize to lowercase. I'll keep it as is, unless you feel strongly.

We spoke offline about the `,`. `,dotnettool;` would be invalid since the thing after the comma is the package type version, not name.